### PR TITLE
Remove all unused code specified by linter warnings

### PR DESCRIPTION
**Pull request by pjlast**
Removes all unused code as highlighted by `sg lint`

## Test plan

Unit tests and integration tests should still pass. My local `sg start` is still behaving as normal.

### CODE REVIEW

**Review by Strum355 — APPROVED** [review]

**Review by mrnugget — APPROVED** [review]

File: enterprise/internal/insights/resolvers/insight_view_resolvers.go [hunk 1]
```diff
@@ -395,11 +395,6 @@ func (r *insightRepositoryDefinitionResolver) ToRepositorySearchScope() (graphql
 
 }
 
-type allReposScope struct {
-}
-
-func (a *allReposScope) AllRepos() bool { return true }
```
Comment (coury-clark) [inline enterprise/internal/insights/resolvers/insight_view_resolvers.go:401]: cc @chwarwick do we still need this for the graphql bindings? I'm not up to date on what we are doing with the all repos state

Comment (chwarwick) [reply enterprise/internal/insights/resolvers/insight_view_resolvers.go:401]: @coury-clark  that one got removed since we knew it was going away.

**Review by sashaostrikov — COMMENTED** [review]
Nice!

**Comment by vdavid** [comment]
This is so cool! <3

**Pull request merged**